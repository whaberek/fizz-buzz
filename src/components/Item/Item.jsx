import styled from 'styled-components';

import { colors } from 'shared/constants';

const Item = styled.span`
  color: ${({ color }) => (color ? color : colors.black)}
`;

export { Item };
