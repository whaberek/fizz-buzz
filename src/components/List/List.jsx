import React, { useState } from 'react';

import { Button, ListItem } from 'components';

import { texts } from 'shared/constants';

const List = ({ data }) => {
  const [currentPage, setCurrentPage] = useState(1);

  const _paginate = (array, page) => array.slice(20 * (page - 1), 20 * page);

  let paginateItems = _paginate(data, currentPage);

  const _nextPage = () => {
    setCurrentPage(currentPage + 1);
    paginateItems = _paginate(data, currentPage);
  };

  const _previousPage = () => {
    setCurrentPage(currentPage - 1);
    paginateItems = _paginate(data, currentPage);
  };

  const _keyExtractor = (item, index) => `${item.props.children}_${index}`;

  return (
    <>
      <>
        <Button type="button" onClick={_previousPage} label={texts.previous}/>
        <Button type="button" onClick={_nextPage} label={texts.next}/>
      </>
      {
        paginateItems.map((item, index) => (
          <ListItem key={_keyExtractor(item, index)}>{item}</ListItem>
        ))
      }
    </>
  );
};

export { List };
