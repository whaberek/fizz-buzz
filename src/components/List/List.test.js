import React from 'react';
import { shallow } from 'enzyme';

import { List } from './List';

import { Item } from 'components/Item/Item'

describe('List', () => {
  const props = {
    data: [
      <Item>foo</Item>,
      <Item>bar</Item>,
      <Item>baz</Item>,
    ],
  };

  it('should render layout', () => {
    const wrapper = shallow(<List {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render 3 ListItem items when listItems contains 3 elements', () => {
    const wrapper = shallow(<List {...props} />);
    expect(wrapper.find(Item)).toHaveLength(3);
  });
});
