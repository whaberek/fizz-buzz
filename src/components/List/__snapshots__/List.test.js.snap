// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should render layout 1`] = `
<Fragment>
  <Button
    label="Previous"
    onClick={[Function]}
    type="button"
  />
  <Button
    label="Next"
    onClick={[Function]}
    type="button"
  />
  <ListItem
    key="foo_0"
  >
    <styled.span>
      foo
    </styled.span>
  </ListItem>
  <ListItem
    key="bar_1"
  >
    <styled.span>
      bar
    </styled.span>
  </ListItem>
  <ListItem
    key="baz_2"
  >
    <styled.span>
      baz
    </styled.span>
  </ListItem>
</Fragment>
`;
